src/ui/atoms/listItem/ListItem.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import type { DeepReadonly } from 'superTypes'
+import { Typography } from '../typography/Typography'
+import './listItem.scss'
+import { isString } from 'utils'
+
+export type ListItemProps = {
+  /**
+   * The main title of the item.
+   */
+  readonly title?: string
+  /**
+   * Additional information about the item
+   */
+  readonly description?: string | JSX.Element
+  /**
+   * An that can provide visual information or allow interactions on the left side.
+   */
+  readonly leftElement?: JSX.Element
+  /**
+   * An that can provide visual information or allow interactions on the right side.
+   */
+  readonly rightElement?: JSX.Element
+}
+
+export const ListItem: React.FC<ListItemProps> = ({
+  title,
+  description,
+  leftElement,
+  rightElement
+}: DeepReadonly<ListItemProps>): JSX.Element => {
+  const renderDescription = (): string | JSX.Element | undefined =>
+    isString(description) ? (
+      <Typography as="div" color="text" fontSize="small">
+        {description}
+      </Typography>
+    ) : (
+      description
+    )
+
+  return (
+    <div className="okp4-listitem-main">
+      {leftElement && <div className="okp4-listitem-left-element">{leftElement}</div>}
+      {title && (
+        <div className="okp4-listitem-title">
+          <Typography as="div" color="text" fontSize="small" fontWeight="bold">
+            {title}
+          </Typography>
+        </div>
+      )}
+      {description && <div className="okp4-listitem-description">{renderDescription()}</div>}
```

Comment:
I think you can move assertion and the `div` container directly in the `renderDescription` function to be more readable, what do you think?